canonical-web-and-design/maas-ui
# Build AddKVMForm.

## Done

- Built AddKVMForm and AddKVMFormField components.
- Updated PowerTypeFields component and formatPowerParameters util to account for "pod" driver type as well as "node" and "chassis"

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- This can be hard to QA without real hardware, so maybe it's enough to check that the network request is good:
- Go to /MAAS/r/kvm and check that there is an "Add KVM" button (and an inactive action menu)
- Click "Add KVM" and check that you are directed to /MAAS/r/kvm/add
- Check that a couple of the form fields change depending on the selected KVM host type
- Fill in the form, submit and take note of the network request. You should get an error, unless you're doing this for real.
- Go to /MAAS/l/kvm and add a KVM there with the same form values
- Check that the network request is the same, apart from a few parameters that don't need to be included (check [the api](https://maas.io/docs/api#pods) to see what's required)

## Fixes

Fixes canonical-web-and-design/MAAS-squad#1954

## Screenshots
### /r/kvm
![0 0 0 0_8400_MAAS_r_kvm (5)](https://user-images.githubusercontent.com/25733845/85366455-b5d4bb80-b56a-11ea-8e6c-4b798ed12d14.png)

### /r/kvm/add
![0 0 0 0_8400_MAAS_r_kvm (6)](https://user-images.githubusercontent.com/25733845/85366459-bb320600-b56a-11ea-9d8c-522864592cc4.png)
